### PR TITLE
HZ-8 ActiveAgent API, setAgentToActive in AgentService class

### PR DESCRIPTION
**Pull request by Bjia0008**
without test

### CODE REVIEW

File: src/main/java/com/eta/houzezbackend/controller/AgentController.java [hunk 1]
```diff
@@ -29,6 +28,12 @@ public AgentGetDto getAgent(@PathVariable Long id) {
         return agentService.getAgent(id);
     }
 
+
+    @PatchMapping("/verification")
```
Comment (yuanningd) [inline src/main/java/com/eta/houzezbackend/controller/AgentController.java:32]: /id

File: src/test/java/com/eta/houzezbackend/service/AgentServiceTest.java [hunk 2]
```diff
@@ -95,12 +105,21 @@ void shouldGetActiveLinkWhenCreateSignUpLink(){
         String baseUrl = "base";
         String id = "id";
         String name = "name";
-        Integer minute = 10;
+        int minute = 10;
         String jwt = "jwt";
         when(jwtService.createJWT(id,name,minute)).thenReturn(jwt);
-        assertEquals(agentService.createSignUpLink(baseUrl,id,name,minute), baseUrl + "/verification?code=" + jwt);
+        assertEquals(agentService.createSignUpLink(baseUrl,id,name,minute), baseUrl + "/agents/decode/" + jwt);
```
Comment (yuanningd) [inline src/test/java/com/eta/houzezbackend/service/AgentServiceTest.java:111]: Keep the old one

File: src/main/java/com/eta/houzezbackend/controller/AgentController.java [hunk 3]
```diff
@@ -29,8 +29,8 @@ public AgentGetDto getAgent(@PathVariable Long id) {
     }
 
 
-    @PatchMapping("/verification")
-    public Agent activeAgent(@RequestParam String token){
+    @PatchMapping("/verification/{id}")
```
Comment (yuanningd) [inline src/main/java/com/eta/houzezbackend/controller/AgentController.java:32]: Think about the meaning behind restful API, do we need "verification" here? ` @PatchMapping("/{id}")`

File: src/test/java/com/eta/houzezbackend/service/AgentServiceTest.java (hunk 2 above)
Comment (yuanningd) [reply src/test/java/com/eta/houzezbackend/service/AgentServiceTest.java:111]: And what is this base url? Base URL for fronend or backend. We need frontend btw

Comment (Bjia0008) [reply src/test/java/com/eta/houzezbackend/service/AgentServiceTest.java:111]: > And what is this base url? Base URL for fronend or backend？ We need frontend btw

So I need to change the base-url param from http://localhost:8080/api/v1 to frontend url in the .yml file, is that correct?

Comment (yuanningd) [reply src/test/java/com/eta/houzezbackend/service/AgentServiceTest.java]: yes

File: src/test/java/com/eta/houzezbackend/controller/AgentControllerTests.java [hunk 4]
```diff
@@ -62,4 +74,18 @@ void shouldReturn404WhenCannotFindByEmail() throws Exception {
...
+
+    @Test
+    void shouldReturn200AndAgentWhenSetAgentToActive() throws Exception{
+        mockMvc.perform(patch("/agents/" + mockUserId +"?token=" + mockJwt))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.activated").value(true));
+    }
+
+    @Test
+    void shouldReturn401WhenSetAgentToActive() throws Exception{
```
Comment (yuanningd) [inline src/test/java/com/eta/houzezbackend/controller/AgentControllerTests.java:86]: Good jon

**Review by yuanningd — APPROVED** [review]

**Pull request merged**